.github/workflows/arc-mlir.yml

```diff
@@ -35,10 +35,9 @@ jobs:
       run: |
         cd arc-mlir
         ./arc-mlir-build
-        for COMMIT in $COMMITS; do
+        for COMMIT in $(git rev-list mlir..HEAD); do
```

Comment:
Instead of doing the loop in this script, wouldn't a cleaner version be to trigger a separate workflow doing the test of each commit? My thinking is that this action which triggers on a push only runs rev-list and then triggers a separate manually activated work flow for each commit.

Have a look at https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch it looks like you would have to mess around with access tokens and stuff, but it looks like it would be possible,

Reply:
👍 I will take a look

Reply:
I asked on the GitHub actions board and they just sent a response: https://github.community/t/execute-job-for-all-commits-in-pull-request/125140

Reply:
I'm not sure how github will handle the results, but if you get a test-line for each commit it will be easier to see which commit failed.

Reply:
This one seems relevant https://github.community/t/run-a-workflow-for-a-specific-commit-sha/18303/10

Reply:
and https://github.com/marketplace/actions/workflow-dispatch

Reply:
So I guess I need to 1) Fetch the GitHub commits, 2) extract the SHA:s, 3) Post a `workflow_dispatch` events for every SHA.